### PR TITLE
Added schedule_search.dart

### LINKED ISSUE #39: Schedule search -- middleware
Define a `ScheduleSearch` model with the following: 

- [x] `List<Subject> getMatchingClasses(String query)`: Returns classes that match a search query 
- [x] `List<String> getPeriods(Subject subject)`: Returns the periods (user-friendly names) when a subject meets

### PR DESCRIPTION
**Pull request by todesj**
fixes #39 
Blocked by #64

### CODE REVIEW

File: lib/src/models/view/schedule_search.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import "package:ramaz/data.dart";
+import "package:ramaz/models.dart";
+
+extension on Subject{
+  String get id => "";
+}
```
Comment (Levi-Lesches) [inline lib/src/models/view/schedule_search.dart:6]:
```suggestion

```

This was a small "hack" to stop the error. But we do need to actually properly fix this. I'll let Brayden know

**Review by Levi-Lesches — CHANGES_REQUESTED** [review]
1. Doc comments everywhere
2. Did you (or your IDE) run `dart format` or `flutter format`? I think we should stay away from that for this project. You can add some spaces between things like the type arguments for `MapEntry`, `index=0`, etc.  Also, check that the indentation is tabs and not spaces

File: lib/src/models/view/schedule_search.dart (hunk 1 above)
Comment (todesj) [reply lib/src/models/view/schedule_search.dart:6]: yeah i know i just didnt want to get an error when i pushed

File: lib/src/models/view/schedule_search.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    subjects = model.subjects.values.toList();
+  }
+
+  List<Subject> getMatchingClasses(String query) =>
+      [
+        for(final Subject subject in subjects)
+          if(subject.name.contains(query))
+            subject
+      ];
+  List<String> getPeriods(Subject subject){
```
Comment (Levi-Lesches) [inline lib/src/models/view/schedule_search.dart:18]:
```suggestion
  List<Subject> getMatchingClasses(String query) => [
    for (final Subject subject in subjects)
      if (subject.name.contains(query))
        subject
  ];
  
  List<String> getPeriods(Subject subject){
```

Some formatting

**Review by Levi-Lesches — APPROVED** [review]

**Comment by Levi-Lesches** [comment]
Not merging, blocked by #64

**Review by DavidTarrab — CHANGES_REQUESTED** [review]
Me and Brayden were talking this through, and we think you should add getMatchingTeachers and getMatchingIDs as well.

**Pull request merged**